Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/EnumDataType.java

```diff
@@ -106,7 +106,17 @@ public String getName(long value) {
 
 	@Override
 	public String[] getNames() {
-		return nameMap.keySet().toArray(new String[nameMap.size()]);
+		Comparator<String> compareByValue = new Comparator<String>() {
+			@Override
+			public int compare(String arg0, String arg1) {
+				Long a0 = getValue(arg0);
+				Long a1 = getValue(arg1);
+				return a0.compareTo(a1);
+			}
+		};
+		String[] values = nameMap.keySet().toArray(new String[nameMap.size()]);
+		Arrays.sort(values, compareByValue);
+		return values;
```

Comment:
Instead of creating a new string array of the names and then sorting them via the comparator you can use a stream and `Map.Entry::comparingByValue`. It may just be personal preference though.

It could look something like this
```
@Override
public String[] getNames() {
    return nameMap.entrySet().stream()
                             .sorted(Map.Entry.comparingByValue())
                             .map(Map.Entry::getKey)
                             .toArray(String[]::new);
}
```

Reply:
Thanks, I figured there would be a better java/oop way.